src/main/java/com/mineboundteam/minebound/magic/OffensiveSpells/EnderOffensiveSpell.java

```diff
@@ -33,21 +34,14 @@ public EnderOffensiveSpell(Properties properties, EnderOffensiveSpellConfig conf
     @Override
     public void use(ItemStack stack, Level level, Player player) {
         if (!level.isClientSide()) {
-            float direction = player.getYRot();
-            double x = (0 - Math.sin(direction * Math.PI / 180f));
-            double z = (Math.cos(direction * Math.PI / 180f));
-            int distance = 1;
-            while (distance <= teleportDistance) {
-                BlockPos tpLocation = new BlockPos(player.getPosition(1).add(distance * x, 0, distance * z));
-                // use isPossibleToRespawnInThis function to check if block has collision
-                if (level.getBlockState(tpLocation).getBlock().isPossibleToRespawnInThis()) {
-                    distance++;
-                } else {
-                    break;
-                }
-            }
-            if (--distance > 0) {
-                player.teleportTo(player.getX() + distance * x, player.getY(), player.getZ() + distance * z);
+            BlockHitResult result = (BlockHitResult) player.pick(teleportDistance, 1f, false);
```

Comment:
Does this work on air blocks too? Or do they have to be looking at a block? I like being able to teleport directly upwards if I need to

Reply:
Yup, works on air blocks.